v3/src/components/case-tile-common/inspector-panel/ruler-menu-list.tsx

```diff
@@ -1,17 +1,30 @@
+import { useDisclosure } from "@chakra-ui/react"
 import { observer } from "mobx-react-lite"
-import React from "react"
+import React, { useState } from "react"
 import { useDataSetContext } from "../../../hooks/use-data-set-context"
 import { logStringifiedObjectMessage } from "../../../lib/log-message"
 import { IAttribute } from "../../../models/data/attribute"
 import { createAttributesNotification } from "../../../models/data/data-set-notifications"
 import { uiState } from "../../../models/ui-state"
+import { convertDatasetToCsv } from "../../../utilities/csv-export"
+import { initiateImportFromCsv } from "../../../utilities/csv-import"
 import { uniqueName } from "../../../utilities/js-utils"
 import { preventCollectionReorg } from "../../../utilities/plugin-utils"
 import { t } from "../../../utilities/translation/translate"
+import { CopiedCasesAlert } from "../copied-cases-alert"
+import { CopyToClipboardModal } from "../copy-to-clipboard-modal"
 import { IMenuItem, StdMenuList } from "../std-menu-list"
 
 export const RulerMenuList = observer(function RulerMenuList() {
   const data = useDataSetContext()
+  const [copiedCasesString, setCopiedCasesString] = useState("")
+  const copyToClipboardModal = useDisclosure()
+  const copiedAlert = useDisclosure()
```

Comment:
This is the usage we're moving away from. Dereference the return immediately:
```suggestion
  const { isOpen: isCopyModalOpen, onClose: onCloseClipboardModal, onOpen: onOpenClipboardModal } = useDisclosure()
  const { isOpen: isCopiedAlertOpen, onClose: onCloseCopiedAlert, onOpen: onOpenCopiedAlert } = useDisclosure()
```